refresh TopSQL docs and screenshots (#22559)
pingcap/docs: dashboard/top-sql.md

- After you enable Top SQL, this feature continuously collects CPU workload data from existing TiDB and TiKV nodes and retains the data for up to 30 days.
- Starting from v8.5.6 and v9.0.0, you can also enable **TiKV Network IO collection (multi-dimensional)** in the Top SQL settings to further view metrics such as `Network Bytes` and `Logical IO Bytes` for specified TiKV nodes, and perform aggregation analysis in dimensions of `By Query`, `By Table`, `By DB`, and `By Region`.

Top SQL provides the following features:

* Visualize the top `5`, `20`, or `100` SQL queries with the most resource consumption in the current time range through charts and tables, with the remaining records automatically summarized as `Others`.
* Display resource consumption hotspots sorted by CPU time or network bytes. When selecting a TiKV node, you can also sort by logical IO bytes.
* Display SQL and execution plan details by query. When selecting a TiKV node, you can also aggregate analysis in dimensions of `By Table`, `By DB`, and `By Region`.

> * Support zooming the chart by selecting a time range, manual refresh, auto refresh, and exporting to CSV.

Reviewer: ```suggestion
* Zoom in on a selected time range in the chart, manually refresh data, enable auto refresh, and export table data to CSV.
```

* Collect all SQL statements that are executed, including those that are still running.

> * Allow you to view data of a specific TiDB and TiKV node.

Reviewer: ```suggestion
* Display data of a specific TiDB or TiKV node.
```

## Use Top SQL

The following are the common steps to use Top SQL.

1. Visit the [Top SQL page](#access-the-page).

2. Select a particular TiDB or TiKV node that you want to observe the workload.

    ![Select a TiDB or TiKV node](/media/dashboard/v8.5-top-sql-usage-select-instance.png)

    If you are not sure which node to observe, you can first locate the node with abnormal workload from Grafana or the [TiDB Dashboard Overview page](/dashboard/dashboard-overview.md), and then return to the Top SQL page for further analysis.

3. Set the time range and refresh data as needed.

    You can adjust the time range in the time picker or zoom the observation window by selecting a time range in the chart. Setting a smaller time range displays more fine-grained data, with a precision of up to 1 second.

    ![Change time range](/media/dashboard/v8.5-top-sql-usage-change-timerange.png)

    If the chart is out of date, click **Refresh** to refresh once, or select the data auto-refresh frequency from the **Refresh** drop-down list.

    ![Refresh](/media/dashboard/v8.5-top-sql-usage-refresh.png)

4. Select the observation mode.

>     - Use `Limit` to select the Top `5`, `20`, or `100` SQL queries to display.

Reviewer: ```suggestion
    - Use `Limit` to display the Top `5`, `20`, or `100` SQL queries.
```

    - The default aggregation dimension is `By Query`. If you select a TiKV node, you can also aggregate in dimensions of `By Table`, `By DB`, or `By Region`.

        ![Select aggregation dimension](/media/dashboard/v8.5-top-sql-usage-select-agg-by.png)

>     - The default sort order is `Order By CPU` (sorted by CPU time). If you select a TiKV node and have [enabled TiKV Network IO collection (multi-dimensional)](#enable-tikv-network-io-collection-optional), you can also select `Order By Network` (sorted by network bytes) or `Order By Logical IO` (sorted by logical IO bytes).

Reviewer 2: ```suggestion
    - The default sort order is `Order By CPU` (sorted by CPU time). If you select a TiKV node and have [enabled TiKV Network IO collection (multi-dimensional)](#optional-enable-tikv-network-io-collection-new-in-v856-and-v900), you can also select `Order By Network` (sorted by network bytes) or `Order By Logical IO` (sorted by logical IO bytes).
```

        ![Select order by](/media/dashboard/v8.5-top-sql-usage-select-order-by.png)

    > **Note**
    >

>     > `By Region` and `Order By Network`, `Order By Logical IO` are only available when [TiKV Network IO collection (multi-dimensional)](#enable-tikv-network-io-collection-optional) is enabled. If this feature is not enabled but historical data still exists, the page will continue to display historical data and prompt that new data cannot be fully collected.

Reviewer 2: ```suggestion
    > `By Region` and `Order By Network`, `Order By Logical IO` are only available when [TiKV Network IO collection (multi-dimensional)](#optional-enable-tikv-network-io-collection-new-in-v856-and-v900) is enabled. If this feature is not enabled but historical data still exists, the page will continue to display historical data and prompt that new data cannot be fully collected.
```

    - TiDB nodes usually show `Call/sec` and `Latency/call`.
    - TiKV nodes usually show `Call/sec`, `Scan Rows/sec`, and `Scan Indexes/sec`.

    > **Note**
    >
    > If you select the `By Table`, `By DB`, or `By Region` aggregation view, the page displays the aggregation results and does not show SQL statement details by SQL execution plan.

    In the `By Query` view, you can also click **Search in SQL Statements** in the Top SQL table to jump to the corresponding SQL Statement Analysis page. If you need to analyze the current table results offline, you can click **Download to CSV** above the table to export the current table data.

7. On TiKV nodes, if you need to locate hotspots from a higher dimension, you can switch to `By Table`, `By DB`, or `By Region` to view the aggregated results.

    ![Aggregated results at DB level](/media/dashboard/v8.5-top-sql-usage-agg-by-db-detail.png)

8. Based on these initial clues, you can further analyze the root cause using the [SQL Statement](/dashboard/dashboard-statement-list.md) or [Slow Queries](/dashboard/dashboard-slow-query.md) page.

## Disable Top SQL

You can disable this feature by following these steps:

> 1. Visit [Top SQL page](#access-the-page).

Reviewer: ```suggestion
1. Visit the [Top SQL page](#access-the-page).
```

2. Click the gear icon in the upper right corner to open the settings pane and disable the **Enable Feature** switch.
3. Click **Save**.
4. In the popped-up dialog box, click **Disable**.

> After disabling, new Top SQL data collection will stop, but historical data can still be viewed before it expires.

Reviewer: ```suggestion
After you disable Top SQL, new Top SQL data collection will stop, but historical data can still be viewed before it expires.
```

**2. Will performance be affected after enabling Top SQL?**

Enabling Top SQL has a slight impact on cluster performance. According to measurements, the average performance impact is less than 3%. If you also enable TiKV Network IO collection (multi-dimensional), there will be additional storage and query overhead.

**3. What is the status of this feature?**

It is now a generally available (GA) feature and can be used in production environments.

**4. What does `Others` mean in the UI?**

> `Others` represents the summary result of all non-Top N records under the current sort dimension. You can use this result to understand the proportion of Top N records in the overall workload.

Reviewer: ```suggestion
`Others` represents the summary result of all non-Top N records under the current sort dimension. You can use it to understand how much of the total workload comes from the Top N records.
```

**5. What is the relationship between the CPU overhead displayed by Top SQL and the actual CPU usage of the process?**

Their correlation is strong but they are not exactly the same thing. For example, the cost of writing multiple replicas is not counted in the TiKV CPU overhead displayed by Top SQL. In general, SQL statements with higher CPU usage result in higher CPU overhead displayed in Top SQL.

**6. What does the Y-axis of the Top SQL chart mean?**

The Y-axis of the Top SQL chart represents the resource consumption under the current sort dimension.

- When `Order By CPU` is selected, the Y-axis represents CPU time.
- When `Order By Network` is selected, the Y-axis represents network bytes.
- When `Order By Logical IO` is selected, the Y-axis represents logical IO bytes.

**7. Does Top SQL collect running (unfinished) SQL statements?**

> Yes. TiDB Dashboard collects the resource consumption of all running or completed SQL after Top SQL is enabled, so unfinished SQL is also collected.

Reviewer: ```suggestion
Yes. After you enable Top SQL, TiDB Dashboard collects resource consumption for all running SQL statements, including unfinished ones.
```
